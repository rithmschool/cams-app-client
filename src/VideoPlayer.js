import React, {Component} from 'react';
import YouTube from 'react-youtube';

class VideoPlayer extends Component {
  constructor(props){
    super(props);
    this._onEnd = this._onEnd.bind(this)
  }

  _onReady(e) {
    // access to player in all event handlers via event.target
    e.target.playVideo();
  }

  _onEnd(e){
    e.target.stopVideo();
    this.props.toggle();
  }

  componentDidMount(){
    this.props.toggle();
  }

  render() {
    const opts = {
      height: '562.5',
      width: '1000',
      playerVars: { // https://developers.google.com/youtube/player_parameters
        autoplay: 1,
        controls: 0,
        rel: 0
      }
    };
    return (
      <div>
        <YouTube
          videoId={this.props.videoId}
          opts={opts}
          onReady={this._onReady}
          onEnd={this._onEnd}
        />
        <p className="lg">Press the space key to move on once the video has ended </p>
      </div>
    )
  }
}

export default VideoPlayer;
