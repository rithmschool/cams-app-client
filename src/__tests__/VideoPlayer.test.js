import VideoPlayer from "../VideoPlayer";
import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import toJson from "enzyme-to-json";

test("matches snapshot of VideoPlayer", () => {
  const videoPlayer = renderer
    .create(<VideoPlayer toggle={function() {}} videoId={"something"} />)
    .toJSON();
  expect(videoPlayer).toMatchSnapshot();
});

test("should render with text ", () => {
  const videoPlayer = shallow(
    <VideoPlayer toggle={function() {}} videoId={"something"} />
  );
  expect(videoPlayer.find(".lg")).toHaveLength(1);
});

test("component mounts ", () => {
  const spy = jest.spyOn(VideoPlayer.prototype, "componentDidMount");
  const wrapper = shallow(
    <VideoPlayer toggle={function() {}} videoId={"something"} />
  );
  wrapper.instance().componentDidMount();
  expect(spy).toHaveBeenCalled();
});
