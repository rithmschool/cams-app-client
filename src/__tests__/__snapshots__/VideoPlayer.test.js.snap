// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of VideoPlayer 1`] = `
<div>
  <span>
    <div
      className={undefined}
      id={undefined}
    />
  </span>
  <p
    className="lg"
  >
    Press the space key to move on once the video has ended
     
  </p>
</div>
`;
